src/contract/client.ts

```diff
@@ -12,20 +12,79 @@ import { processSpecEntryStream } from './utils';
 const CONSTRUCTOR_FUNC = "__constructor";
 
 async function specFromWasm(wasm: Buffer) {
-  const wasmModule = await WebAssembly.compile(wasm);
-  const xdrSections = WebAssembly.Module.customSections(
-    wasmModule,
-    "contractspecv0"
-  );
-  if (xdrSections.length === 0) {
+  const customData = parseWasmCustomSections(wasm);
```

Comment:
I would rather use the native method if it's available, with this as a fallback.

Reply:
Makes sense to me. I can make this a try..catch